### DISCUSSION TITLE
Versioning logic?

### DISCUSSION THREAD

**Discussion by Noah231515** [immich-app/immich#2636] [Q&A] [answered]
Hey all,

What is the current versioning logic?
I see there are breaking changes from v1.58.0 to v1.59.0 at least under semantic versioning, that would require a new major version.

This caught my attention when upgrading my installation. I upgraded only minor versions and did not expect breaking changes, and my installation broke. Partly my fault to be fair for not reading the changelogs.

Though it would be nice if it followed semantic versioning, since breaking changes are not expected across minor versions.

Anyone have any thoughts/input on this?
Thanks!

**Comment by alextran1502** [comment, marked as answer]
We are still in a fast development cycle, so the versioning is to keep track of the progress/iteration of the project. When a stable release is reached (2?), then any breaking change would require more proper major version changes

**Reply by Noah231515** [reply]
Got it, I'll be sure to pay attention to the changelogs next time.

Thanks for all the hard work!
